src/nvim/ex_cmds.c

```diff
@@ -5721,14 +5720,7 @@ void ex_sign(exarg_T *eap)
             }
 
             xfree(sp->sn_text);
-            // Allocate one byte more if we need to pad up
-            // with a space.
-            len = (int)(p - arg + ((cells == 1) ? 1 : 0));
-            sp->sn_text = vim_strnsave(arg, len);
-
-            if (cells == 1) {
-              STRCPY(sp->sn_text + len - 1, " ");
-            }
+            sp->sn_text = vim_strnsave(arg, p - arg);
```

Comment:
This is in a separate commit - not strictly necessary, but allows for having signs only take one cell after all.